vars/buildPluginWithGradle.groovy

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env groovy
+
+/**
+ * Simple wrapper step for building a plugin with Gradle.
+ */
+def call(Map params = [:]) {
+    // Faster build and reduces IO needs
+    properties([
+        durabilityHint('PERFORMANCE_OPTIMIZED'),
+        buildDiscarder(logRotator(numToKeepStr: '5')),
+    ])
+
+    def repo = params.containsKey('repo') ? params.repo : null
+    def failFast = params.containsKey('failFast') ? params.failFast : true
+    def timeoutValue = params.containsKey('timeout') ? params.timeout : 60
+    if(timeoutValue > 180) {
+      echo "Timeout value requested was $timeoutValue, lowering to 180 to avoid Jenkins project's resource abusive consumption"
+      timeoutValue = 180
+    }
+
+    boolean publishingIncrementals = false
+    boolean archivedArtifacts = false
+    Map tasks = [failFast: failFast]
+    buildPlugin.getConfigurations(params, true).each { config ->
+        String label = config.platform
+        String jdk = config.jdk
+        String jenkinsVersion = config.jenkins
+        String javaLevel = config.javaLevel
+
+        String stageIdentifier = "${label}-${jdk}${jenkinsVersion ? '-' + jenkinsVersion : ''}"
+        
+        tasks[stageIdentifier] = {
+            node(label) {
+                timeout(timeoutValue) {
+                    // Archive artifacts once with pom declared baseline
+                    boolean doArchiveArtifacts = !jenkinsVersion && !archivedArtifacts
```

Comment:
`jenkinsVersion` is a string as is for example `javaLevel`. Sometimes like here its absence is checked via Groovy truth and sometimes via a `null` check. That's inconsistent and might lead to subtile problems.

Reply:
Copied from the original script